### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by blueberry1312**
## In this PR, We did the following changes:

- Preserve (save and load) books data using json files.
- Preserve (save and load) people data using json files.
- Preserve (save and load) rentals data using json files.

### CODE REVIEW

File: people_m.rb [hunk 1]
```diff
@@ -37,14 +39,41 @@ def create_person
...
     else
       puts 'Please enter a valid number (1 or 2)'
       create_person
     end
   end
+
+  def save_people_to_json
+    data = @people.map(&:to_h)
+    File.write('people.json', JSON.generate(data))
+  end
```
Comment (Gambit142) [inline people_m.rb:56]: - [x] Great work implementing this feature. Currently, the data is saved to the _**JSON**_ file during the running of the application. The requirements state that data should be saved to the _**JSON**_ file on exit from your application. Kindly configure your code in such a way that the _**methods**_ that save the data to the _**JSON**_ files will be called on exit of the application. Please implement this for the methods that save people, books and rentals to the _**JSON**_ files. 😎 💻 

image

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @blueberry1312 @Cabraham1 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Created methods for saving data to JSON files ✔️ 
- Created methods for loading data from JSON files ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: people_m.rb (hunk 1 above)
Comment (blueberry1312) [reply people_m.rb:56]: Oh yes We missed that, thanks!

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @blueberry1312 & @Cabraham1, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- All linters checks passed ✔️
- Created methods for saving data to JSON files ✔️
- Created methods for loading data from JSON files ✔️
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**